graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -736,30 +812,55 @@ def _add_edge_field(
             )
         )
 
-    new_edge_field_node = ast_types.FieldDefinition(
-        name=ast_types.Name(value=new_edge_field_name),
+    new_edge_field_node = ast_types.FieldDefinitionNode(
+        name=ast_types.NameNode(value=new_edge_field_name),
         arguments=[],
-        type=ast_types.ListType(
-            type=ast_types.NamedType(name=ast_types.Name(value=sink_type_name),),
+        type=ast_types.ListTypeNode(
+            type=ast_types.NamedTypeNode(name=ast_types.NameNode(value=sink_type_name),),
         ),
         directives=[_build_stitch_directive(source_field_name, sink_field_name),],
     )
 
-    type_fields.append(new_edge_field_node)
+    new_type_fields = list(type_fields)
+    new_type_fields.append(new_edge_field_node)
+    if type(source_type_node) == ast_types.ObjectTypeDefinitionNode:
+        new_source_type_node = ast_types.ObjectTypeDefinitionNode(
+            description=source_type_node.description,
+            name=source_type_node.name,
+            directives=source_type_node.directives,
+            fields=new_type_fields,
+            interfaces=source_type_node.interfaces,
+        )
+    elif type(source_type_node) == ast_types.InterfaceTypeDefinitionNode:
```

Comment:
This might be another good candidate for use of `to_kwargs()` — it might help merge the object/interface branches into a single branch.